src/nccl_ofi_sendrecv.cpp

```diff
@@ -500,13 +512,14 @@ static int sendrecv_recv_conn_post(nccl_net_ofi_sendrecv_listen_comm_t *l_comm,
 				   nccl_net_ofi_sendrecv_ep_t *ep,
 				   void *buffer,
 				   size_t size,
-				   struct fid_mr *mr,
 				   nccl_net_ofi_sendrecv_req_t *req)
 {
 	ssize_t rc = 0;
 	int ret = 0;
 	int dev_id = l_comm->base.base.dev_id;
-	void *desc = fi_mr_desc(mr);
+	auto *freelist_handle = 
+		reinterpret_cast<sendrecv_freelist_mr_handle_t *>(l_comm->conn_info->mr_handle);
```

Comment:
I called this out earlier, but there's no reason to have a `sendrecv_freelist_mr_handle_t`, and it makes this bit of code really ugly.  Just have a single mr handle type, and have the `mr_handle` field in the `freelist_elem_t` be a pointer to a struct of that type, as is the `mhandle` passed into `deregMr`.